website/views/shared/tasks/index.jade

```diff
@@ -6,7 +6,7 @@ include ./task_view/mixins
 script(id='templates/habitrpg-tasks.html', type="text/ng-template")
   .tasks-lists.container-fluid
     .row
-      .col-md-3.col-sm-6(ng-repeat='list in lists', ng-class='::{"rewards-module": list.type==="reward"}')
+      .col-md-3.col-sm-6(ng-repeat='list in lists track by $index', ng-class='::{"rewards-module": list.type==="reward"}')
```

Comment:
$index in this case should be entirely unrelated to the data model. It
exists gives the ngRepeat directive a way of distinguishing each node in
the dom and in a way that's unique to each list. I'll defer to your
judgment on the tasks part, are there other places where it doesn't make
sense? I'll clean up the PR to target more specific places if so.
On Jun 26, 2015 8:52 PM, "Blade Barringer" <notifications@github.com> wrote:

> In website/views/shared/tasks/index.jade
> <https://github.com/HabitRPG/habitrpg/pull/5500#discussion_r33408751>:
>
> > @@ -6,7 +6,7 @@ include ./task_view/mixins
> >  script(id='templates/habitrpg-tasks.html', type="text/ng-template")
> >    .tasks-lists.container-fluid
> >      .row
> > -      .col-md-3.col-sm-6(ng-repeat='list in lists', ng-class='::{"rewards-module": list.type==="reward"}')
> > +      .col-md-3.col-sm-6(ng-repeat='list in lists track by $index', ng-class='::{"rewards-module": list.type==="reward"}')
>
> Until the task migration is complete, we cannot use track by $index
> because some tasks have duplicate ids and Angular breaks if that happens
> while track by is turned on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/HabitRPG/habitrpg/pull/5500/files#r33408751>.
>